packages/flutter_background_service_android/android/src/main/java/id/flutter/flutter_background_service/WatchdogReceiver.java

```diff
@@ -66,10 +66,14 @@ public void onReceive(Context context, Intent intent) {
             }
 
             if (!config.isManuallyStopped() && !isRunning) {
-                if (config.isForeground()) {
-                    ContextCompat.startForegroundService(context, new Intent(context, BackgroundService.class));
-                } else {
-                    context.startService(new Intent(context, BackgroundService.class));
+                try {
+                    if (config.isForeground()) {
+                        ContextCompat.startForegroundService(context, new Intent(context, id.flutter.flutter_background_service.BackgroundService.class));
+                    } else {
+                        context.getApplicationContext().startService(new Intent(context, id.flutter.flutter_background_service.BackgroundService.class));
+                    }}
+                catch (Exception e){
+                    e.printStackTrace();
```

Comment:
What are the exception/error we are trying to catch here and root cause?

Reply:
Root cause for this is line no. 73
When context.getApplicationContext() is called sometimes it returns null. Due to which crash is happening